### PR TITLE
Introduce NoOp connector

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Depends on https://github.com/prestosql/presto/pull/2596

**Comment by martint** [comment]
What’s the motivation? Why not use the blackhole or memory connectors instead?

**Comment by kokosing** [comment]
> Why not use the blackhole or memory connectors instead?

Blackhole and memory connectors contain dependencies and so they are a bit more expensive to load.

**Comment by findepi** [comment]
> What’s the motivation? Why not use the blackhole or memory connectors instead?

@martint for product tests we need to create dummy connector config files (side effect of using docker; also present before Lauuncher, we just ignored the problem)

I need a connector that is really really noop -- it must not support create table, for example.
(maybe it could be named `empty`)


> Depends on #2596

@kokosing let me know when this becomes independent
that's not high priority, so i will defer the review

**Comment by kokosing** [comment]
@findepi updated

File: presto-noop/src/main/java/io/prestosql/plugin/memory/NoOpPlugin.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        @Override
+        public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+        {
+            return new NoOpMetadata();
+        }
+    }
+
+    private static class NoOpMetadata
+            implements ConnectorMetadata {}
```
Comment (findepi) [inline presto-noop/src/main/java/io/prestosql/plugin/memory/NoOpPlugin.java]: I think the transaction handle should be defined the same way

File: presto-noop/src/test/java/io/prestosql/plugin/memory/NoOpQueryRunner.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    public static DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Logging.initialize();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner
+                .builder(testSessionBuilder().build())
+                .setNodeCount(4)
```
Comment (findepi) [inline presto-noop/src/test/java/io/prestosql/plugin/memory/NoOpQueryRunner.java]: redundant

**Pull request merged**